core/lib/Session.ts

```diff
@@ -80,6 +80,7 @@ export default class Session extends TypedEventEmitter<{
     if (options.userProfile) {
       this.userProfile = options.userProfile;
       this.browserEmulator.userProfile = options.userProfile;
+      this.browserEmulator.sessionId = this.id;
```

Comment:
I think we likely want to set sessionId even if there's not a userProfile, don't we?

Reply:
Agreed. Pushed fix.